src/html/index.html

```diff
@@ -1,11 +1,36 @@
 <!doctype html>
 <html lang="en">
-  <head>
-    <meta charset="utf-8">
-    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
-    <title><%= htmlWebpackPlugin.options.title || 'Webpack App'%></title>
-  </head>
-  <body>
-    <div id="root"></div>
-  </body>
+
+<head>
+  <%if (htmlWebpackPlugin.options.ipfsHack) { %>
+  <script>
+    (function () {
+      const { origin, pathname } = window.location
+      const ipfsMatch = /.*\/Qm\w{44}/.exec(pathname)
+      const base = document.createElement('base')
+      let baseHref
+      if (ipfsMatch) {
+
+        baseHref = origin + ipfsMatch[0]
```

Comment:
Why is the origin required, cannot be relative for example to the domain root?

For example:
* **For url**: https://gateway.ipfs.io/ipfs/QmeYYwD4y4DgVVdAzhT7wW5vrvmbKPQj8wcV2pAzjbj886/docs/getting-started/
* **Base href would be**:`/ipfs/QmeYYwD4y4DgVVdAzhT7wW5vrvmbKPQj8wcV2pAzjbj886/`

Reply:
Not required, no real reason I included it.